### ISSUE TITLE
[Hoedown comeback regression] Code in doc not tested with future warning, when it was always a test in the past

### ISSUE THREAD

**Issue by kennytm** [rust-lang/rust#41401]
Test case:

```rust
// src/lib.rs

/// ```
/// assert!(true);
/// ```
///
/// ```
/// assert!(true);
/// ```
pub fn test() {}
```

**Expected:** when running `cargo test`, it will run two doctests (as in stable and beta).

```
running 2 tests
test src/lib.rs - test (line 1) ... ok
test src/lib.rs - test (line 5) ... ok
```

**Actual:** Only the first doctest is run, the second one is skipped, with this warning:

```
WARNING: src/lib.rs - test (line 5) Code block is not currently run as a test, but will in future versions of rustdoc. Please ensure this code block is a runnable test, or use the `ignore` directive.

running 1 test
test src/lib.rs - test (line 1) ... ok
```

Real life test case: 

* Code: https://github.com/iron/params/blob/8f9db17a4265264327f010fadb6e2231e0e53bde/src/lib.rs#L214-L237
* Doc: https://docs.rs/params/0.6.0/params/struct.Map.html#examples
* Travis: https://travis-ci.org/iron/params/jobs/223656157#L421-L429

(cc @GuillaumeGomez #41290)

----

No warnings in this version:

```
rustdoc 1.18.0-nightly (28a742997 2017-04-13)
binary: rustdoc
commit-hash: 28a74299778cdad4ea999e4ee8f8c1ef793338bd
commit-date: 2017-04-13
host: x86_64-apple-darwin
release: 1.18.0-nightly
LLVM version: 3.9
```

Have warnings in this version:

```
rustdoc 1.18.0-nightly (9f2abadca 2017-04-18)
binary: rustdoc
commit-hash: 9f2abadca2d065bf81772cb84981d0a22d8e98b3
commit-date: 2017-04-18
host: x86_64-apple-darwin
release: 1.18.0-nightly
LLVM version: 3.9
```

**Comment by steveklabnik** [comment]
Interesting; it _is_ parsed as two separate blocks by CommonMark: http://spec.commonmark.org/dingus/?text=%60%60%60%0Aassert!(true)%3B%0A%60%60%60%0A%0A%60%60%60%0Aassert!(true)%3B%0A%60%60%60

Hoedown is not on https://johnmacfarlane.net/babelmark2/?text=%60%60%60%0Aassert!(true)%3B%0A%60%60%60%0A%0A%60%60%60%0Aassert!(true)%3B%0A%60%60%60 but it looks like mostly trivial variations

**Comment by kennytm** [comment]
@steveklabnik It *is* rendered as two separate blocks in Hoedown, in stable as well as current nightly (9f2abadca). It is just that rustdoc now becomes too conservative in classifying code blocks as tests.

**Comment by steveklabnik** [comment]
@kennytm yes, sorry. It looks like this is a hoedown bug; @GuillaumeGomez is on the case

**Comment by GuillaumeGomez** [comment]
Problem comes from the fact that hoedown gives an invalid line number so I have to fix code we want to remove. T_T

**Comment by dtolnay** [comment]
Anyone know a workaround for this? Are there any magic changes I can make to my doc comment for now to get both blocks to run?

**Comment by kennytm** [comment]
@dtolnay Wait for the next nightly where the bug is fixed 😆

**Comment by kennytm** [comment]
@dtolnay You could add an explicit paragraph break between the code blocks e.g.

```rust
/// ```
/// assert!(true);
/// ```
///
/// <!-- -->
///
/// ```
/// assert!(true);
/// ```
```

But I'd just wait for the next nightly which has this bug fixed.

**Comment by kennytm** [comment]
Additional test cases:

1. ```rust
    /// a
    ///
    /// ```
    /// assert!(true);
    /// ```
    ///
    /// b
    ///
    /// ```
    /// assert!(true);
    /// ```
    pub fn test() {}
    ```

2. ```rust
    /// | Table |
    /// |-------|
    /// | Value |
    ///
    /// ```rust
    /// assert!(true);
    /// ```
    pub fn test() {}
    ```

See https://github.com/rust-lang/rust/pull/41405#issuecomment-295551295 for details.
